Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,9 +133,49 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
+            // Skip null and read-only parameters
             if (parameter == null || parameter.IsReadOnly)
                 return false;
 
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
+            {
+                // Skip empty values
+                if (string.IsNullOrWhiteSpace(value as string))
+                    return false;
+
+                // Find all user worksets
+                FilteredWorksetCollector worksets =
+                    new FilteredWorksetCollector(document)
+                    .OfKind(WorksetKind.UserWorkset);
+
+                // Does a workspace with the specified name exist?
+                foreach (Workset workset in worksets)
+                {
+                    if (workset.Name == value as string)
+                    {
+                        // Workset found
+                        // Switch the dropdown menu to the required workset based on its index number
+                        int worksetIndex = workset.Id.IntegerValue; // Index number of the existing workset
+                        return parameter.Set(worksetIndex); // Select a dropdown list item
+                    }
+                }
+
+                // Workset doesn't exist
+                if (document.IsWorkshared) // Skip non-workshared documents
```

Comment:
This check should happen at the beginning

Reply:
Moved to the top